apps/mobile/src/hooks/useBiometrics.ts

```diff
@@ -7,85 +7,217 @@ import {
+   */
+  const checkOSBiometricsSupport = useCallback(async () => {
     try {
-      const supportedBiometrics = await Keychain.getSupportedBiometryType()
-
-      if (supportedBiometrics) {
-        let type: 'FACE_ID' | 'TOUCH_ID' | 'FINGERPRINT' | 'NONE' = 'NONE'
-
-        switch (supportedBiometrics) {
-          case Keychain.BIOMETRY_TYPE.FACE_ID:
-            type = 'FACE_ID'
-            break
-          case Keychain.BIOMETRY_TYPE.TOUCH_ID:
-            type = 'TOUCH_ID'
-            break
-          case Keychain.BIOMETRY_TYPE.FINGERPRINT:
-            type = 'FINGERPRINT'
-            break
-        }
+      // First check what type of biometry is supported by hardware
+      const biometryType = await DeviceCrypto.getBiometryType()
 
-        dispatch(setBiometricsType(type))
-        dispatch(setBiometricsSupported(true))
-        return true
-      }
+      // If the device reports biometrics support, we need to verify enrollment
+      if (biometryType !== BiometryType.NONE) {
+        Logger.info('Device reports biometrics hardware support:', biometryType)
 
-      return false
-    } catch (error) {
-      Logger.error('Error checking biometrics support:', error)
-      return false
-    }
-  }, [dispatch])
+        // Use a unique service name for our test to avoid conflicts
+        const biometricTestService = '__BIOMETRIC_ENROLLMENT_TEST__'
 
-  const checkBiometricsOSSettingsStatus = useCallback(async () => {
-    try {
-      // This checks if biometrics is available at system level
-      const result = await Keychain.getSupportedBiometryType()
-      // If biometrics is not set up at OS level, this will return null
-      // If Face ID is available, it returns 'FaceID'
-      // If Touch ID is available, it returns 'TouchID'
+        try {
+          // First try to clear any existing test entry to start fresh
+          await Keychain.resetGenericPassword({ service: biometricTestService })
+
+          // On iOS, we need to specify BIOMETRY_ANY to test biometric enrollment
+          // On Android, we use BIOMETRY_CURRENT_SET which is more accurate for enrollment check
+          const accessControl =
+            Platform.OS === 'ios' ? Keychain.ACCESS_CONTROL.BIOMETRY_ANY : Keychain.ACCESS_CONTROL.BIOMETRY_CURRENT_SET
+
+          // Step 1: Try to set a password with biometric protection
+          Logger.info('Testing biometric enrollment - setting test password')
+          await Keychain.setGenericPassword('biometric_test', 'enrollment_test_value', {
+            service: biometricTestService,
+            accessControl,
+            accessible: Keychain.ACCESSIBLE.WHEN_UNLOCKED,
+          })
+
+          // Step 2: Immediately try to retrieve it with biometrics
+          // If biometrics aren't enrolled, this will fail with a specific error
+          // If we get here without errors, biometrics are enrolled and working
+          Logger.info('Testing biometric enrollment - retrieving test password')
```

Comment:
the comments and the code make no sense? How are we retrieving? I just see a log?